pandas/core/arrays/datetimelike.py

```diff
@@ -396,6 +396,34 @@ def floor(self, freq, ambiguous="raise", nonexistent="raise"):
     def ceil(self, freq, ambiguous="raise", nonexistent="raise"):
         return self._round(freq, RoundTo.PLUS_INFTY, ambiguous, nonexistent)
 
+    def _with_freq(self, freq):
```

Comment:
can you type

Reply:
Optional[Union[Dateoffset, str]]

Reply:
returns Union[DTA,TDI,PA] (but I think we have a type for this)

Reply:
just tried this and it looks like its going to turn into a yak-shaving event, prefer to do that separately

Reply:
kk, followup if you can